src/Aspire.Hosting/ApplicationModel/ResourceUrlAnnotation.cs

```diff
@@ -30,4 +30,24 @@ public sealed class ResourceUrlAnnotation : IResourceAnnotation
     /// The display order the URL. Higher values mean sort higher in the list.
     /// </summary>
     public int? DisplayOrder;
+
+    /// <summary>
+    /// Whether this URL should be shown on the dashboard resources page. Defaults to <see langword="true"/>.
+    /// </summary>
+    /// <remarks>
+    /// Set this to <see langword="false"/> to only show this URL in the resource details grid.
+    /// </remarks>
+    public bool ShowOnResourcesPage { get; set; } = true;
```

Comment:
Thoughts on the property name? This maps to the `IsInternal` property on the URL snapshot and resource server APIs but that name is pretty terrible so trying to make it more descriptive here.

Reply:
make it an enum?

Reply:
Ha I actually started that way, a flags enum.

Reply:
Actually not flags as they dashboard and protocol currently only support the boolean state (`IsInternal`). So perhaps something like the following, although I admit I don't love it:

```csharp
/// <summary>
/// Specifies where the URL should be displayed.
/// </summary>
public enum UrlDisplayLocation
{
    /// <summary>
    /// Show the URL in locations where either the resource summary or resource details are being displayed.
    /// </summary>
    SummaryAndDetails,
    /// <summary>
    /// Show the URL in locations where the full details of the resource are being displayed.
    /// </summary>
    DetailsOnly
}
```

Reply:
Flags

Reply:
And modify the protocol?

Reply:
Maybe flags is wrong anyways because alll combinations aren’t valid so the above is fine?